### ISSUE TITLE
Live samples in translated-content cannot use images from content

### ISSUE THREAD

**Issue by SphinxKnight** [mdn/yari#5652]
### Summary

The images used in live sample (`{{EmbedLiveSample}}`) are not served (404) when they are not duplicated on mdn/translated-content.

### URL

https://developer.mozilla.org/fr/docs/Web/CSS/filter

### Reproduction steps

1. Go to https://developer.mozilla.org/fr/docs/Web/CSS/filter
2. Checkout mdn/translated-content as of [e1b9e2c](https://github.com/mdn/translated-content/commit/e1b9e2c7998ffd7c7dc01c1de496b7842934abd5) and a local yari
3. Compare live samples in the page

### Expected behavior

Both (local and production) should match (at least after some time)

### Actual behavior

In production https://yari-demos.prod.mdn.mozit.cloud/fr/docs/Web/CSS/filter/test_form_3.jpeg is 404

### Device

Laptop

### Browser

Firefox

### Browser version

Stable

### Operating system

Windows

### Screenshot

![image](https://user-images.githubusercontent.com/2413436/158791939-7a67dc2d-294e-4850-bd36-87d464c8dec4.png)

### Anything else?

Previously https://github.com/mdn/yari/pull/5215 was merged, it worked locally, I waited a bit and made https://github.com/mdn/translated-content/pull/4549

https://github.com/mdn/translated-content/issues/3752

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mdn/yari/discussions).

**Comment by yin1999** [comment]
I have an idea about this problem, is it easier to build with **correct image path**? It means that if the image is not in `relative path` at `translated-content` repo but in the `content` repo, we just replace the `relative path` with the `absolute path` (to `content` repo) before writing to `index.html` file.

This doesn't complicate the proxy any more (proxy is needed to check whether the image file exists in `en-US` when `S3` report that the file is not exists in `l10n`).

My english is not very well, apologies.

**Comment by SphinxKnight** [comment]
@mdn/core-yari-dev is it possible to have a look at this one? (ex. https://developer.mozilla.org/fr/docs/Web/CSS/filter still failing at loading resources)

**Comment by yin1999** [comment]
I found two function which may help with this.

- `checkImageReferences` in `build\check-images.js`: is used to fallback the `src` attribute of `img` to `en-US` sources while the src does not exist in `l10n`.
- `buildLiveSamplePages` in `kumascript\src\live-sample.js`: is used to build `_sample_.some_id.html`, but it will not check the `image reference`

Note that, livesample in https://developer.mozilla.org/fr/docs/Web/CSS/filter does have element `image` with attribute `xlink:href`.

**Comment by SphinxKnight** [comment]
@mdn/core-dev should I revert https://github.com/mdn/yari/pull/5215? is there anything we can do here?

**Comment by caugner** [comment]
@SphinxKnight We believe that #5215 is not the cause. It just doesn't seem to work for live samples (yet), and so removing the corresponding images from the translated-content repository caused the images to no longer show.

For now, the best solution would be to restore those images in translated-content.

**Comment by yin1999** [comment]
I have a new idea about resolving this problem.

Could we just generate a json file to list paths of files in mdn/content (exclude `.md` and `.html` file). just like #6124.

So we can use aws-lambda to redirect (302 redirect) the request of image file to `en-US` by modifying the response. Examples on [aws](https://docs.aws.amazon.com/en_us/AmazonCloudFront/latest/DeveloperGuide/lambda-examples.html#lambda-examples-custom-error-new-site).

The condition we may use: 

- **status code is 404** (file not exists in l10n)
- **the requested file name (exclude `.md` and `.html`) exists in the JSON file**

And I found a answer about image redirection on [stackoverflow](https://stackoverflow.com/a/3778511/12349560).

**Comment by schalkneethling** [comment]
Friendly ping on this one. @SphinxKnight, @yin1999, @caugner should we open a discussion to explore options here?

**Comment by yin1999** [comment]
> should we open a discussion to explore options here?

Agree, we should gather more ideas.

**Comment by yin1999** [comment]
A new idea about this:

We are now hosting about 11 locales (9 after [Archiving pl and de locales](https://github.com/orgs/mdn/discussions/147)). And we may modify the request **before** finding the file hosted on aws-S3 (so we can reduce redirects for those images which have to fallback to `en-US`, the method metioned [above](https://github.com/mdn/yari/issues/5652#issuecomment-1136689773))

So, we may go on this way (similar to the [redirect logic of lambda@edge](https://github.com/mdn/yari/pull/6124)):

1. collect all those image paths in `mdn/content` repo
2. for each image path, check whether it's in l10n folders. If not, flag it.
3. store those flags of image path in a file
4. in [Origin request](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-cloudfront-trigger-events.html), check whether the request is to l10n folder. If so, and the filepath has been flaged, modify the `request.uri` to `en-us` folder.

The problem is, there are too many images that are not in most l10n folders:

> for about 1800 non-md files, we should create about 18000 flags (about 10 times) for those files are not in l10n.

So this may not be a good idea to store each not-existed l10n image file path. (if there are 4000 image files in mdn/content in the future, we should store about 40000 filepaths).

Another way to flag those file is `bitwise` flag (just like the [Unix Octal permissions](https://tylercipriani.com/blog/2020/01/12/unix-permissions-for-dummies/)):

For each locale, we use a `powers of 2` number to present. For example:

| locale | number | bitwise |
| ------ | -------- | -------- |
| en-us | 1          | 0000001 |
| zh-cn | 2          | 0000010 |
| zh-tw | 4          | 0000100 |
| fr       | 8          | 0001000 |

If `path/to/image` does not exist in `fr` and `zh-cn`, we can compress the flags by sum up all the numbers: `2` (for zh-cn) + `8` (for fr) = `10`.

In [Origin request](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-cloudfront-trigger-events.html), we can:

- Check whether the `path/to/image` exists in `fr`: `10` (for sumed flags) & `8` (for fr) = 1. The result is not zero, so the image is not in `fr`.
- Check whether the `path/to/image` exists in `zh-tw`: `10` (for sumed flags) & `4` (for zh-tw) = 0. The result is zero, so the image is in `zh-tw`.

For the sumed flags, we can just use a map to present (easy to stored as a json):

```js
{
  “path/to/image_1”: 10,
  "path/to/image_2": 8,
  // etc.
}
```

And in JavaScript, the bitwise operation is safe for 32-bit integer, so we can host 32 locales at most.

**Comment by caugner** [comment]
We have finally discussed this is in the engineering meeting today, and we will proceed by copying the referenced content images next to the translated-content during build.

**Comment by yin1999** [comment]
> We have finally discussed this is in the engineering meeting today, and we will proceed by copying the referenced content images next to the translated-content during build.

Maybe this issue could also be fixed mdn/translated-content#7648. 🤗

**Comment by caugner** [comment]
> Maybe this issue could also be fixed [mdn/translated-content#7648](https://github.com/mdn/translated-content/issues/7648). 🤗

Yeah, copying the images over (as part of the build!) also for images outside of live samples would probably fix that.

(The alternative would be to always also build the `en-US` version of a modified page in the PR Companion.)

**Comment by Graywolf9** [comment]
Hello to everyone!

If it helps and about this issue: https://github.com/mdn/translated-content/issues/12130

It shows the images correctly if I run

from content repo (http://localhost:5042/es/docs/Web/CSS/object-fit):
```
yarn start
```

from yari repo (http://localhost:3000/es/docs/Web/CSS/object-fit):
```
yarn dev
```

![Screenshot 2023-03-13 at 12-46-29 MDN Web Docs](https://user-images.githubusercontent.com/13079269/224800243-df3e2168-9f62-4c77-90d0-2378fc99fa18.png)

**Comment by Graywolf9** [comment]
About my above comment I noticed that:

1. In the deployed site, it's looking for the images at `yari-demos.prod.mdn.mozit.cloud/[lang]`
2. In `content`, `yarn start`, it's doing what @caugner says [here](https://github.com/mdn/yari/issues/5652#issuecomment-1346348334) and looking for the images at `localhost:5042/[lang]` or what it's the same `developer.mozilla.org/[lang]`
3. In `yari`, `yarn dev`, it's doing the same as `content` and looking for the images at `localhost:3000/[lang]` or what it's the same `developer.mozilla.org/[lang]`
4. Also I noticed that in the deployed site in `en-US` the images are in `yari-demos.prod.mdn.mozit.cloud/en-US` and at the same time in `developer.mozilla.org/en-US` (before this error appears all was pointing to `developer.mozilla.org`)
